File: two-pointers/11. Container With Most Water.js
/**
 * @param {number[]} height
 * @return {number}
 */
var maxArea = function (height) {
  let left = 0, right = height.length - 1;
  let max = 0;
  
  while (left < right) {
    let minHeight = Math.min(height[left], height[right]);
    max = Math.max(max, minHeight * (right - left));
    
    if (height[left] < height[right]) {
      left++;
    } else {
      right--;
    }
  }
  return max;
};

// Test
const height = [1, 8, 6, 2, 5, 4, 8, 3, 7]; //49
// const height = [1,1]; //1
console.log('maxArea: ', maxArea(height)); // eslint-disable-line no-console

File: python/two_pointers/11. Container With Most Water.py
from typing import List

class Solution:
    def maxArea(self, height: List[int]) -> int:
        left = 0
        right = len(height) - 1
        max_height = 0

        while left < right:
            min_height = min(height[left], height[right])
            max_height = max(max_height, min_height * (right - left))

            if height[left] < height[right]:
                left += 1
            else:
                right -=1

        return max_height

sol = Solution()

# heights = [1,8,6,2,5,4,8,3,7] # 49
heights = [1,1] #1
print(f'sol: {sol.maxArea(heights)}')
